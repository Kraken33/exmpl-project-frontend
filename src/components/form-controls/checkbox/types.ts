import { CheckboxProps } from "antd/lib/checkbox";
import { FormItemProps } from "antd/lib/form";
import { WrappedFieldProps } from "redux-form";

export type FControlCheckbox = {
  type?: "input" | "textarea";
} & FormItemProps &
  WrappedFieldProps &
  CheckboxProps;
