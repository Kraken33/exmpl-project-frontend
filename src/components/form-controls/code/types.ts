import { WrappedFieldProps } from 'redux-form';
import { FormItemProps } from 'antd/lib/form';

export interface FControlCode extends FormItemProps, WrappedFieldProps {
    type?: 'input' | 'textarea';
};