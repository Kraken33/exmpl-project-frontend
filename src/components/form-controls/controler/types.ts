import { FormItemProps } from 'antd/lib/form';

export enum ETypes {
    'input' = 'input',
    'switch' = 'switch',
    'code' = 'code'
};

export type Types = keyof typeof ETypes;

export interface Dictionary {
    [k: string]: any;
}

export type Control = {
    type: Types;
} & FormItemProps;