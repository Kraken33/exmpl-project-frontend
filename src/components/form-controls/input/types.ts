import { FormItemProps } from "antd/lib/form";
import { InputProps, TextAreaProps } from "antd/lib/input";
import { WrappedFieldProps } from "redux-form";

export type FControlInput = FormItemProps &
  InputProps &
  TextAreaProps &
  WrappedFieldProps & {
    type?: "input" | "textarea";
  };
