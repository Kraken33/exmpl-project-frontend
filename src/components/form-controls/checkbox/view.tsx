import { Checkbox } from "antd";
import FormItem from "components/form-item";
import React from "react";

import { FControlCheckbox } from "./types";

const Component: React.FC<FControlCheckbox> = ({
  input,
  meta,
  type,
  label,
  required,
  ...attributes
}) => {
  return (
    <FormItem required={required} label={label} {...meta}>
      <Checkbox
        onChange={input.onChange}
        checked={input.value}
        {...attributes}
      />
    </FormItem>
  );
};

export { Component };
