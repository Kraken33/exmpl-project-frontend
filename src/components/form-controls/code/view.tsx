import React from 'react';
import { Controlled as CodeMirror } from "react-codemirror2";

import { FControlCode } from './types';

import FormItem from 'components/form-item';

const Component: React.FC<FControlCode> = ({ input, meta, type, label, required, ...attributes }) => {
  return (
    <FormItem required={required} label={label} {...meta}>
      <CodeMirror
        {...attributes}
        options={{
          theme: 'material',
          lineNumbers: true
        }}
        value={input.value}
        onBeforeChange={(_, __, value) => {
          input.onChange(value);
        }}
      />
    </FormItem>
  );
}

export { Component };