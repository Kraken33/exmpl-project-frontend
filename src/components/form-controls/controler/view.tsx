import React from "react";
import { Field } from "redux-form";
import { error } from "utils/errors";

import { Checkbox } from "../checkbox";
import { Code } from "../code";
import { Input } from "../input";
import { Switch } from "../switch";
import { Control, Dictionary, Types } from "./types";

const Component: React.FC<Control> = ({ type, ...props }) => {
  const typeSwitcher = (componentTypeName: Types) => {
    const typesDictionary: Dictionary = {
      input: Input,
      password: Input,
      switch: Switch,
      code: Code,
      checkbox: Checkbox,
    };
    const currentFormElement = typesDictionary[componentTypeName];

    !currentFormElement &&
      error(`Type ${componentTypeName} missing in form control types!`);

    return currentFormElement || Input;
  };

  return (
    <Field {...(props as any)} type={type} component={typeSwitcher(type)} />
  );
};

export { Component };
