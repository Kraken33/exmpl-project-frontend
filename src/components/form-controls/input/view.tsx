import React from 'react';
import { Input } from "antd";
import FormItem from 'components/form-item';

import { FControlInput } from './types';

const { TextArea } = Input;

const Component: React.FC<FControlInput> = ({ input, meta, type, label, required, ...attributes }) => {
  return (
    <FormItem required={required} label={label} {...meta}>
      {type === "textarea" ? (
        <TextArea {...input} {...attributes as any} />
      ) : (
          <Input {...input} {...attributes as any} type={type} />
        )}
    </FormItem>
  );
}

export { Component };