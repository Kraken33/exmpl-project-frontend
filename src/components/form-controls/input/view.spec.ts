import { shallow } from 'enzyme';
import { r } from 'utils/component-test';

import { Component } from './view';
import { FControlInput } from './types';

describe('testing input type form control', ()=>{
    const props: FControlInput = {
        input: {
            value: '',
            onChange: ()=>null
        } as any,
        meta: {
            touched: false,
            error: false,
            valid: true
        } as any,
        label: 'Label',
        required: false
    };

    it('match snapshot', ()=>{
        const wrapper = shallow(r<FControlInput>(Component, props));

        expect(wrapper).toMatchSnapshot();
    });
});