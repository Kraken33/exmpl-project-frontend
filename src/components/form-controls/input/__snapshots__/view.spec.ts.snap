// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing input type form control match snapshot 1`] = `
<Component
  error={false}
  label="Label"
  required={false}
  touched={false}
  valid={true}
>
  <Input
    onChange={[Function]}
    type="text"
    value=""
  />
</Component>
`;
